typedb/common/exception.py

```diff
@@ -25,7 +25,7 @@
 
 class TypeDBClientException(Exception):
 
-    def __init__(self, msg: Union["ErrorMessage", str], cause: Optional[BaseException], params: Any = None):
+    def __init__(self, msg: Union["ErrorMessage", str], cause: BaseException = None, params: Any = None):
```

Comment:
Given that Python is so easy to misuse, we may as well make the class a little more resilient against misuse.